### PR TITLE
[mini] Initialize profiler in mini_init

### PR DESCRIPTION
**Pull request by lambdageek**
Fixes segfault when cooperative GC is enabled.

### CODE REVIEW

**Comment by lambdageek** [comment]
Attn: @vargaz

File: mono/mini/driver.c [hunk 1]
```diff
@@ -2006,6 +2020,8 @@ mono_main (int argc, char* argv[])
 	/* Set rootdir before loading config */
 	mono_set_rootdir ();
 
+	mini_threads_runtime_init ();
+
 	if (enable_profile) {
```
Comment (vargaz) [inline mono/mini/driver.c]: Still don't understand why profiling loading can't be a normal part of runtime initialization.

Comment (lambdageek) [inline mono/mini/driver.c]: You mean separate the profiler option parsing and profiler initialization, do the option stuff here, and the initialization in `mono_init_internal`?  I could give that a shot.

Comment (vargaz) [inline mono/mini/driver.c]: Can we just save the profiler options somewhere, and call the profile load in mini_init ().

**Comment by kumpera** [comment]
Hey @alexrp, did you review this one?

**Pull request merged**